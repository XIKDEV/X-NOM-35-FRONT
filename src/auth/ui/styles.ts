import { ILoginButton, ILoginInput } from '../../interfaces';

export const loginInput: ILoginInput = {
  boxShadow: 'none',
  borderColor: 'var(--primary)',
};

export const loginButton: ILoginButton = {
  color: 'var(--gray)',
};
