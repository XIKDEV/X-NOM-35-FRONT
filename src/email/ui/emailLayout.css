.email-layout {
  width: 50%;
  border-radius: 4px;
  padding: 48px 24px;
  box-sizing: border-box;

  background: var(--gray);
}
.email-layout img {
  width: min(50%, 154px);
}
.email-layout h1 {
  font-size: var(--title);
  color: var(--primary);
}
.email-text p {
  font-size: var(--text);
}
.email-text .password {
  margin: 16px 0px;
}
.email-text .password p {
  margin: 0px;
}
.email-text .password p:first-child {
  margin-bottom: 12px;
  font-size: var(--subtitle-small);
  color: var(--primary);
}
.email-text .password p:last-child {
  font-size: var(--title-small);
}
.email-text a {
  padding: 12px 24px;
  margin: 16px 0px;
  border-radius: 4px;

  background-color: var(--primary);

  color: var(--gray);
  text-decoration: none;
}
.email-text p span {
  font-size: var(--subtext-small);
}

/** Laptop screen **/
@media (min-width: 769px) {
  .email-layout {
    width: min(50%, 800px);
    border-radius: 8px;
  }
  .email-text .password {
    margin: 32px 0px;
  }
  .email-text .password p {
    margin: 0px;
  }
  .email-text .password p:first-child {
    margin-bottom: 12px;
    font-size: var(--subtitle);
    color: var(--primary);
  }
  .email-text .password p:last-child {
    font-size: var(--title);
  }
  .email-text a {
    padding: 18px 32px;
    border-radius: 8px;
  }
  .email-text p span {
    font-size: var(--subtext);
  }
}
