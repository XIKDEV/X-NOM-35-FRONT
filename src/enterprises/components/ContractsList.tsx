import { FC } from 'react';
import { Card, List } from 'antd';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFileArrowDown } from '@fortawesome/free-solid-svg-icons';
import { IContractsListProps } from '../../interfaces';
import {
  bodyContractStyle,
  listContractsStyleDesktop,
  listContractsStyleMobile,
  listItemContractsStyle,
} from '../ui';

export const ContractsList: FC<IContractsListProps> = ({ data, isMobile }) => {
  return (
    <List
      style={isMobile ? listContractsStyleMobile : listContractsStyleDesktop}
      dataSource={data}
      renderItem={(item) => (
        <List.Item style={listItemContractsStyle}>
          <Card bodyStyle={bodyContractStyle} className="enterprise-contract">
            <div className="enteprise-contract-container flex-row-center space-between">
              <p>{item.contract}</p>
              <p>{item.date}</p>
              <button>
                <FontAwesomeIcon icon={faFileArrowDown} />
              </button>
            </div>
          </Card>
        </List.Item>
      )}
    />
  );
};
