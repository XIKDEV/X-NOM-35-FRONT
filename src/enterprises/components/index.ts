export * from './ContractsList';
export * from './EnterprisesCard';
export * from './HeadEnterpriseInfo';
export * from './InfoEnterprise';
