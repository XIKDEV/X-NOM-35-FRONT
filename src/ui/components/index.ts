export * from './CardMobile';
export * from './FormItemInput';
export * from './FormItemInputPassword';
export * from './FormItemSelect';
export * from './FormItemSelectCoordinate';
export * from './FormItemUpload';
export * from './FormsButtons';
export * from './FormsDrawer';
export * from './HeaderMobile';
export * from './HeaderPage';
export * from './InfoDrawerButtons';
