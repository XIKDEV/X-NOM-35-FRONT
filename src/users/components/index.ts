export * from './HeaderUserInfo';
export * from './InfoUser';
export * from './UsersCard';
