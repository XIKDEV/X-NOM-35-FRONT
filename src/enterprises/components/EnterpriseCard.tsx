import { FC } from 'react';
import { Card } from 'antd';
import Meta from 'antd/es/card/Meta';
import { ICardProps } from '../../interfaces';
import { enterprisesConstants } from '../../constants';
import { enterprisesCardBodyStyle, enterprisesMetaStyle } from '../ui';

export const EnterpriseCard: FC<ICardProps> = ({
  enterpriseName,
  responsible,
  logoEnterprise,
}) => {
  return (
    <Card
      className="card-enterprise br-8"
      hoverable
      bodyStyle={enterprisesCardBodyStyle}
      cover={
        <img
          alt={`${enterprisesConstants.logoFrom} ${enterpriseName}`}
          src={logoEnterprise}
        />
      }
    >
      <Meta
        title={
          <p className="card-labels" style={enterprisesMetaStyle.title}>
            {enterpriseName}
          </p>
        }
        description={
          <p className="card-labels" style={enterprisesMetaStyle.description}>
            {enterprisesConstants.responsibleEnterprise} {responsible}
          </p>
        }
      />
    </Card>
  );
};
