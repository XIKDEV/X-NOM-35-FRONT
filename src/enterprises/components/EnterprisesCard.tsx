import { FC } from 'react';
import { Card } from 'antd';
import Meta from 'antd/es/card/Meta';
import { ICardProps } from '../../interfaces';
import { cardMetaStyle } from '../../ui/styles';
import { enterprisesCardBodyStyle } from '../ui';

export const EnterprisesCard: FC<ICardProps> = ({
  srcImage,
  title,
  subtitle,
  alt,
  handleOpenDrawerInfo,
}) => {
  return (
    <Card
      className="card-enterprise br-8"
      hoverable
      bodyStyle={enterprisesCardBodyStyle}
      cover={<img src={srcImage} alt={alt} />}
      onClick={handleOpenDrawerInfo}
    >
      <Meta
        title={
          <p className="card-labels" style={cardMetaStyle.title}>
            {title}
          </p>
        }
        description={
          <p className="card-labels" style={cardMetaStyle.description}>
            {subtitle}
          </p>
        }
      />
    </Card>
  );
};
