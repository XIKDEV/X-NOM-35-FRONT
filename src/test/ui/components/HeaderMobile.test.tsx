import { render } from '@testing-library/react';
import { HeaderMobile } from '../../../ui/components';

jest.mock('antd/es/card/Meta', () => {
  return {
    __esModule: true,
    default: () => <div>MockedMetaComponent</div>,
  };
});

describe('Test in <HeaderMobile />', () => {
  test('should be match with the snapshot', () => {
    const handleSidebar = jest.fn();
    const { asFragment } = render(
      <HeaderMobile handleSidebar={handleSidebar} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
