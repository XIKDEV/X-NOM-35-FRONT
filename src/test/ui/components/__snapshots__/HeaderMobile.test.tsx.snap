// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <HeaderMobile /> should be match with the snapshot 1`] = `
<DocumentFragment>
  <header
    class="ant-layout-header header-mobile flex-row-center space-between css-dev-only-do-not-override-17a39f8"
  >
    <button
      data-testid="sidebar-toggle-button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bars "
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
          fill="currentColor"
        />
      </svg>
    </button>
    <input
      class="br-4"
      placeholder="Búsqueda rápida"
      type="text"
    />
    <button>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-plus "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
          fill="currentColor"
        />
      </svg>
    </button>
  </header>
</DocumentFragment>
`;
